src/rust/engine/src/nodes.rs

```diff
@@ -292,16 +292,14 @@ pub struct MultiPlatformExecuteProcess {
 
 impl MultiPlatformExecuteProcess {
   fn lift_process(value: &Value, platform_constraint: Option<Platform>) -> Result<Process, String> {
+    let gil = Python::acquire_gil();
+    let py = gil.python();
     let env = externs::getattr_from_frozendict(value, "env");
-
-    let working_directory = {
-      let val = externs::getattr_as_string(value, "working_directory");
-      if val.is_empty() {
-        None
-      } else {
-        Some(RelativePath::new(val.as_str())?)
-      }
-    };
+    let working_directory =
+      match externs::getattr_as_optional_string(py, value, "working_directory") {
+        None => None,
+        Some(dir) => Some(RelativePath::new(dir)?),
+      };
 
```

Comment:
Just use `.map(|s| RelativePath::new(s)?)`

Reply:
I tried, but complains that the closure isn't returning `Option` or `Result`, so the `?` doesn't work. The only way I could get things to compile is `RelativePath::new().unwrap()`, which is a downgrade from before